### PR TITLE
[LINALG] Add torch.layout information

### PR DESCRIPTION
**Pull request by pashu123**
torch.layout information has been added.

### CODE REVIEW

File: lib/Conversion/TorchToLinalg/TensorConstructors.cpp [hunk 1]
```diff
@@ -181,12 +181,17 @@ class ConvertAtenEmptyMemoryFormatOp
...
 
     // TODO: Add support for layout, pin_memory and memory_format features.
-    // Only `none` layout is supported.
-    if (!op.layout().getType().template isa<Torch::NoneType>())
+    // Only 'strided' and `none` layout is supported. The memory layout should
+    // be `none` or `contiguous` to make sure that default strides are used.
+    int64_t layout;
+    if (!op.layout().getType().template isa<Torch::NoneType>() &&
+        (!matchPattern(op.layout(), m_TorchConstantInt(&layout)) ||
+         layout != torch_upstream::Layout::Strided))
```
Comment (cathyzhyi) [inline lib/Conversion/TorchToLinalg/TensorConstructors.cpp:194]: the last check `layout !=torch_upstream::Layout::Strided` should only be executed when `matchPattern(op.layout(), m_TorchConstantInt(&layout)` succeed. You can refer to the code here:
https://github.com/llvm/torch-mlir/blob/5b9bdfaf3fb56c5aed12d689e21c9884b1c5a789/lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp#L205-L213. There are multiple other places in this file with similar issue. Can you help to fix those as well? Maybe add helps like `checkIfNoneOrConstantbool`, `checkIfNoneOrConstantInt` etc

File: include/torch-mlir/Dialect/Torch/Utils/TorchUpstream.h [hunk 2]
```diff
@@ -97,6 +97,16 @@ enum MemoryFormat {
   ChannelsLast3d
 };
 
+//===----------------------------------------------------------------------===//
+// Possible values for `layout` argument in PyTorch ops that support it.
+// Source:
+// https://pytorch.org/docs/stable/tensor_attributes.html#torch.torch.layout
```
Comment (cathyzhyi) [inline include/torch-mlir/Dialect/Torch/Utils/TorchUpstream.h:103]: I think this is not the right source. Looking at pytorch's implementation of empty here:
https://github.com/pytorch/pytorch/blob/3c1c7fd26e94d00cfa50b7e6dab79f7c7988ae40/aten/src/ATen/EmptyTensor.cpp#L98-L106

I think it's this enum that needs to be copied https://github.com/pytorch/pytorch/blob/3c1c7fd26e94d00cfa50b7e6dab79f7c7988ae40/c10/core/Layout.h#L9

**Review by cathyzhyi — CHANGES_REQUESTED** [review]

File: lib/Conversion/TorchToLinalg/TensorConstructors.cpp (hunk 1 above)
Comment (pashu123) [reply lib/Conversion/TorchToLinalg/TensorConstructors.cpp:194]: Okay sure.

File: lib/Conversion/TorchToLinalg/TensorConstructors.cpp [hunk 3]
```diff
@@ -181,12 +181,17 @@ class ConvertAtenEmptyMemoryFormatOp
...
                   ConversionPatternRewriter &rewriter) const override {
+
     if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
       return failure();
 
     // TODO: Add support for layout, pin_memory and memory_format features.
-    // Only `none` layout is supported.
-    if (!op.layout().getType().template isa<Torch::NoneType>())
+    // Only 'strided' and `none` layout is supported. The memory layout should
+    // be `none` or `contiguous` to make sure that default strides are used.
```
[the remark this answers is not in the document]
Comment (pashu123) [reply lib/Conversion/TorchToLinalg/TensorConstructors.cpp:190]: I see, I will remove the layout check.

File: lib/Conversion/TorchToLinalg/TensorConstructors.cpp (hunk 1 above)
Comment (pashu123) [reply lib/Conversion/TorchToLinalg/TensorConstructors.cpp]: As suggested by sean, I am removing the layout check.

**Review by cathyzhyi — APPROVED** [review]

**Pull request merged**